curriculum/challenges/english/25-front-end-development/quiz-javascript-higher-order-functions/66edcdd18a4ef8df16e6bb7e.md

```diff
@@ -17,439 +17,482 @@ Answer all of the questions below correctly to pass the quiz.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following statements about closures and higher-order functions is true?
 
 #### --distractors--
 
-Placeholder distractor 1
+Higher-order functions must always return a new function; otherwise, they are not considered higher-order.
 
 ---
 
-Placeholder distractor 2
+Closures and higher-order functions are unrelated concepts in JavaScript.
 
 ---
 
-Placeholder distractor 3
+Higher-order functions cannot create closures since they do not return inner functions.
 
 #### --answer--
 
-Placeholder answer
+Closures created by higher-order functions can capture the lexical scope, allowing for state preservation.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Evaluate the Output
+
+```js
+const numbers = [1, 1, 1, 1, 1];
+let sum = 0;
+const forEachRes = numbers.forEach(num => {
+  return (sum += num);
+});
+const mapRes = numbers.map(num => {
+  return (sum += num);
+});
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+forEachRes is undefined and mapRes is [1,2,3,4,5]
 
 ---
 
-Placeholder distractor 2
+forEachRes is 0 and mapRes is [1,2,3,4,5]
 
 ---
 
-Placeholder distractor 3
+forEachRes is 5 and mapRes is [1,2,3,4,5]
 
 #### --answer--
 
-Placeholder answer
+forEachRes is undefined and mapRes is [6,7,8,9,10]
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Evaluate the output
+
+```js
+[, undefined, 'a', 'b', { 20: 5 }].sort();
+```
 
 #### --distractors--
 
-Placeholder distractor 1
+unsupported elements for an array to be sorted, hence error
 
 ---
 
-Placeholder distractor 2
+Callback not supplied, hence error
 
 ---
 
-Placeholder distractor 3
+```js
+[<1 empty item> , 'a', 'b', undefined,{ '20': 5 } ]
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+[ { '20': 5 }, 'a', 'b', undefined, <1 empty item> ]
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following describes a callback function in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+## A function that is called immediately upon declaration.
 
----
-
-Placeholder distractor 2
+A function that is invoked with a specific context.
 
 ---
 
-Placeholder distractor 3
+A function that returns another function.
 
 #### --answer--
 
-Placeholder answer
+A function passed as an argument to another function, to be executed later.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the result of using reduce() on an array?
 
 #### --distractors--
 
-Placeholder distractor 1
+A boolean indicating whether any elements meet a condition.
 
 ---
 
-Placeholder distractor 2
+An Array with all elements reduced by specified callback function
 
 ---
 
-Placeholder distractor 3
+An Array of booleans
 
 #### --answer--
 
-Placeholder answer
+A single accumulated value derived from all elements in the array.
```

Comment:
This is edge case, but not necessarily. Depending on the callback and initial value, it might be ie. array as well.

Reply:
Hey @gikf ,
Yes i agree with it,  The he MDN has it in a separate section, which makes me question whether its out of scope of quiz of this level. 
In case you still want it to be fixed 
Would the below be a better option
'A single accumulated value from the array, or the sole element if the array has one element and no initialValue. Returns undefined for an empty array with no initialValue.'

Reply:
```suggestion
It varies depending on the accumulator's initial value and the callback function.
```